### PR TITLE
211 SIM-OPS-Generate PR from staging to main with changelog

### PR DESCRIPTION
**Pull request by AgustinRamiroDiaz**
Fixes https://github.com/yeagerai/genlayer-simulator/issues/211

# What

- Creates a PR from branches of the form `release-<semver>`, like for example `release-v1.2.3`

# Why

- to automate releases

# Testing done

I've created the branch `release-test` from this branch and
- the GH Action has run successfully https://github.com/yeagerai/genlayer-simulator/actions/runs/10113269538/job/27969244092
- the PR was created successfully https://github.com/yeagerai/genlayer-simulator/pull/346

# Decisions made

I decided to go with branch push as trigger instead of `workflow_dispatch` because I feel it's more comfortable, but I can change that if you think it's more comfortable to do it in the UI

# Checks

- [ ] I have tested this code
- [x] I have reviewed my own PR
- [x] I have created an issue for this PR
- [ ] I have set the PR name to the issue name

# Reviewing tips

GitHub API and CLI docs are very useful

### CODE REVIEW

File: .github/workflows/release-pr.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          echo $latest_release
+          echo "tag=$latest_release" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate release notes
+        id: generate_release_notes
+        # https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release
+        run: |
+          response=$(curl -L \
```
Comment (YeagerAI-Bulat) [inline .github/workflows/release-pr.yml:43]: What if this curl command will fail from some rate-limiting rule? How to handle such errors?

File: .github/workflows/release-pr.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/yeagerai/genlayer-simulator/releases/generate-notes \
+            -d '{"tag_name":"${{ steps.extract_tag.outputs.tag }}","target_commitish":"${{ steps.extract_branch.outputs.branch }}","previous_tag_name":"${{ steps.get_latest_release.outputs.tag}}"}' )
+          echo $response
+
```
Comment (YeagerAI-Bulat) [inline .github/workflows/release-pr.yml:51]: Can you use https://github.com/marketplace/actions/release-notes-generator instead of this?

**Review by YeagerAI-Bulat — CHANGES_REQUESTED** [review]

File: .github/workflows/release-pr.yml (hunk 2 above)
Comment (AgustinRamiroDiaz) [reply .github/workflows/release-pr.yml:51]: First time I see that one, but I've researched a lot about tools to automate this and I didn't find anyone that would satisfy our needs. All of them had little to no flexibility to customize them.

For example, the one you are sharing here only works with "milestones"

Do you know how to use it for this use case?
We need to create release notes when there's a release branch created, and add those notes to a PR body

File: .github/workflows/release-pr.yml (hunk 1 above)
Comment (AgustinRamiroDiaz) [reply .github/workflows/release-pr.yml:43]: This hasn't failed for me in my tests. I think for now we can go forward by just retrying the action manually if it fails and addressing it in a fix PR. I don't think we should complicate it now :)

**Review by cristiam86 — APPROVED** [review]

**Pull request merged**